src/AppBundle/Controller/DashboardController.php

```diff
@@ -0,0 +1,345 @@
+            array_push($componentId, $component->getId());
+            array_push($sizeComponent, $component->getSizeComponent());
+            $forms[$component->getId()] = $this->get('form.factory')->createNamedBuilder($component->getId(), new ComponentType(), $component)->getForm();
+        }
+        //.......Get current dashboard object
+        $em = $this->getDoctrine()->getManager();
+        $dashboard = $em->getRepository('AppBundle:Dashboard')
+            ->find($id);
+        //.......Set dashboard's creator
+        $creator = $dashboard->getCreator();
+
+        //.......Generate form for "Edit Dashboard"
+        $form = $this->get('form.factory')->create(new DashboardType(), $dashboard);
+        $form->handleRequest($request);
+
+        //.......Check if the method is POST and which form component is submitted
+        if ($request->isMethod('POST')) {
+
+            foreach ($forms as $key => $oneForm) {
+                $forms[$key]->handleRequest($request);
+            }
+            foreach ($forms as $key => $oneForm) {
+                if ($oneForm->isSubmitted() && $oneForm->isValid()) {
+
+                    $em = $this->getDoctrine()
+                        ->getManager();
+                    $component = $em->getRepository('AppBundle:Component')
+                        ->find($key);
+                    //.......Change graphic's type
+                    if ($oneForm->get('linechart')->isClicked()) {
+                        $component->setTypeGraph('linechart');
+                    }
+                    if ($oneForm->get('column')->isClicked()) {
+                        $component->setTypeGraph('column');
+                    }
+                    if ($oneForm->get('area')->isClicked()) {
+                        $component->setTypeGraph('area');
+                    }
+                    if ($oneForm->get('bar')->isClicked()) {
+                        $component->setTypeGraph('bar');
+                    }
+                    //.......Save changes made on component
+                    $em->persist($component);
+                    $em->flush();
+
+                    return $this->redirectToRoute('app_dashboard', array(
+                        'id' => $id,
+                        'component' => $component,
+                    ));
+                }
+            }
+        }
+
+        //.......If dashboard edit form was submitted, save changes to database
+        if ($form->isSubmitted() && $form->isValid()) {
+            $em->persist($dashboard);
+            $em->flush();
+
+            return $this->redirectToRoute('app_dashboard', array(
+                'id' => $id));
+        }
+        //.......Create the view for each component form
+        foreach ($forms as $key => $formOne) {
+            $forms[$key] = $formOne->createView();
+        }
+        //.......Render dashboard view and send the variables to it
+        return $this->render('AppBundle:App:dashboard.html.twig', array(
```

Comment:
I deleted some parameters that are not used, the ones that are left are all used in twig ex: `<h2 class="text-center">{{ dashboard.name }}</h2>`